File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        StringBuffer sb = new StringBuffer();        
        for (int i = 1; i < numLog.length; i++)
        {
            int diff = numLog[i] - numLog[i - 1];
            switch (diff)
            {
                case 1 :
                    sb.append('w');
                    break;
                case -1 :
                    sb.append('s');
                    break;
                case 10 :
                    sb.append('d');
                    break;
                case -10 :
                    sb.append('a');
                    break;
            }
        }
        return sb.toString();
    }
}

File: 프로그래머스/Lv.0/181925. 수 조작하기 2/수 조작하기 2.js
function solution(numLog) {
    var answer = '';
    var obj = {
        "1" : "w",
        "-1" : "s",
        "10" : "d",
        "-10" : "a",
    }
    
    for (i = 1; i < numLog.length; i++)
        answer += obj[numLog[i] - numLog[i - 1]];
    return answer;
}